code/datums/elements/noticable_organ.dm

```diff
@@ -43,6 +51,27 @@
 /datum/element/noticable_organ/proc/on_receiver_examine(mob/living/carbon/examined, mob/user, list/examine_list)
 	SIGNAL_HANDLER
 
-	if(body_zone && (body_zone in examined.get_covered_body_zones()))
+	if(!should_show_text(examined))
 		return
-	examine_list += span_notice(replacetext(replacetext("[body_zone ? examined.p_their(TRUE) : examined.p_they(TRUE)] [infused_desc]", "%PRONOUN_ES", examined.p_es()), "%PRONOUN_S", examined.p_s()))
+	var/examine_text = replacetext(replacetext("[body_zone ? examined.p_their(TRUE) : examined.p_they(TRUE)] [infused_desc]", "%PRONOUN_ES", examined.p_es()), "%PRONOUN_S", examined.p_s())
+	if(add_span)
+		examine_text = span_notice(examine_text)
+	examine_list += examine_text
+
+/**
+ * Subtype of noticable organs for AI control, that will make a few more ai status checks before forking over the examine.
+ */
+/datum/element/noticable_organ/ai_control
+	add_span = FALSE
+
+/datum/element/noticable_organ/ai_control/should_show_text(mob/living/carbon/examined)
+	. = ..()
+	if(!.)
+		return FALSE
+	if(examined.ai_controller?.ai_status == AI_STATUS_ON)
+		if(!examined.dna.species.ai_controlled_species)
+			return TRUE
+	return FALSE
+
+/datum/element/noticable_organ/ai_control/on_removed(obj/item/organ/target, mob/living/carbon/loser)
+	Detach(target)
```

Comment:
Why not call parent?

Reply:
because detaching removes the signal anyway. And also, can't put a SIGNAL_HANDLER here, and you'll laugh at the reason why: SIGNAL_HANDLER is actually SHOULD_NOT_SLEEP(TRUE) which is an spacemandmm lint, not related to dreamdaemon. So while SIGNAL_HANDLER is not called here, it is considered because the base proc, `/proc/on_removed` has the lint

Reply:
would've been nice to have it for future protection/clarity but whateva